pages/_app/index.jsx

```diff
@@ -178,18 +200,45 @@ class App extends NextApp {
     return null
   }
 
+  Login = () => {
+    const { router } = this.props
+    const param = 'logout' in router.query ? 'logout' : 'loading'
+
+    return (
+      <iframe
+        ref={this.loginIframeRef}
+        title="Login Form"
+        src={`/login.html?${param}`}
+        className={styles.loginIframe}
+        onLoad={() => {
+          const { isAuthorized, loginInProcess } = this.state
+          if (!isAuthorized && !loginInProcess) {
+            this.loginIframeRef.current.contentWindow.postMessage(
```

Comment:
If you use onLoad iframe event and subscribe from the parent component to it passing a function seems there is no need in posting a message to a window object.

However, if there is a need in that, the `login.js` should post it to own `window`.

Reply:
onLoad event is there just for a case when /user endpoint responds faster than iframe is loaded. 

> However, if there is a need in that, the login.js should post it to own window.

I don't get it. Can you provide some sample code?

Reply:
There is no need in sample. If you post a message from the iframe in the script that is executed in the iframe, the parent should be able to subscribe and get messages.

Also, you can post a message directly to parent using `window.parent.postMessage`.

Reply:
I could not find a case when the [window.parent](https://developer.mozilla.org/en-US/docs/Web/API/Window/parent) is not set but I am sure there is a way to disable it using some [noopener, noreferer policy](https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types). However, I am sure in our case everything should be fine because we use the same origin.

By the way, I found you may use [window.frameOrigin](https://developer.mozilla.org/en-US/docs/Web/API/Window/frameElement) and may even try to fire [custom event](https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent) onto it, e.g. `login` or `logout`.

Reply:
The parent [subscribes](https://github.com/oacore/dashboard/blob/login/pages/_app/index.jsx) to those events.

Also, I use window.parent.postMessage for sending a message to the parent.

The reason why it is like that is that iframe is loaded in the `loading` state, i.e. user sees animation. _app calls /user endpoint in case a user is not logged in the animation is hidden. Onload event handles the case when iframe haven't subscribed to the post message event yet.

PS: line highlighting here in GitHub doesn't work for me. So links will just redirect you to the files.

Reply:
It's not correct to use the login iframe just because to want to show a loading state instead the blank screen. It's 2 different problems and there are 2 different solutions.

Login should solve only its own one.

Reply:
Well, I could extract animation to parent window. I sit that what you propose?


PS: To be honest I see things differently. We've got a problem to solve and there are unlimited ways how to solve it. I don't say something is not correct. It's just a different approach to solve the problem.

Reply:
This, this is what I propose. I agree with P.S., we see different ways of doing this. I am not saying your way is wrong.

Reply:
Ok, I will extract it.

The reason why I am saying it is that sometimes I've got a feeling that you assign me a task where you have a clear idea of how to do it. I do it differently. (Sometimes completely different 😄 sometimes only a little bit). Then you check the code and write different things. I like constructive comments but some of them are too opinionated in my opinion and you kind of insist on them. I consider it wasting time because now two of us work on the same task. You want me to implement your idea, which means I develop something and you review and the cycle repeats. Maybe it would be sometimes better if you take your task. I know you have plenty of other things to do but still. Please don't take it too closely. 🙂

Reply:
You have a good point. I do not mind merging it as it is but I see there is a way to do it better. Your work is really appreciated.

Feel free to point me into the opinionated comments because I may not always notice that.